src/shared/ui/buttons.ts

```diff
@@ -3,25 +3,59 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { QuickInputButton, Uri } from 'vscode'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
 import { ext } from '../extensionGlobals'
+import { WizardControl } from '../wizards/wizard'
+
+const localize = nls.loadMessageBundle()
+const HELP_TOOLTIP = localize('AWS.command.help', 'View Toolkit Documentation')
+
+type WizardButton<T> = QuickInputButton<T | WizardControl> | QuickInputButton<void>
+export type PrompterButtons<T = WizardControl> = readonly WizardButton<T>[]
+
+/** Light wrapper around VS Code's buttons, adding a `onClick` callback. */
+export interface QuickInputButton<T> extends vscode.QuickInputButton {
+    onClick?: () => T
+}
 
 /**
  * Creates a QuickInputButton with a predefined help button (dark and light theme compatible)
  * Images are only loaded after extension.ts loads; this should happen on any user-facing extension usage.
  * button will exist regardless of image loading (UI tests will still see this)
+ *
+ * @param uri Opens the URI upon clicking
  * @param tooltip Optional tooltip for button
  * @param url Optional URL to open when button is clicked
  */
-export function createHelpButton(tooltip?: string, url?: string): HelpButton {
-    return new HelpButton(tooltip, url)
+export function createHelpButton(uri: string | vscode.Uri, tooltip: string = HELP_TOOLTIP): QuickInputLinkButton {
+    const iconPath = {
+        light: vscode.Uri.file(ext.iconPaths.light.help),
+        dark: vscode.Uri.file(ext.iconPaths.dark.help),
+    }
+
+    return new QuickInputLinkButton(uri, iconPath, tooltip)
 }
 
-export class HelpButton implements QuickInputButton {
-    readonly iconPath = {
-        light: Uri.file(ext.iconPaths.light.help),
-        dark: Uri.file(ext.iconPaths.dark.help),
+export class QuickInputLinkButton implements QuickInputButton<void> {
+    public readonly uri: vscode.Uri
+
+    constructor(
+        link: string | vscode.Uri,
+        public readonly iconPath: vscode.QuickInputButton['iconPath'],
+        public readonly tooltip?: string
+    ) {
+        this.uri = typeof link === 'string' ? vscode.Uri.parse(link) : link
     }
 
-    public constructor(readonly tooltip?: string, readonly url?: string) {}
+    public onClick(): void {
+        vscode.env.openExternal(this.uri)
+    }
+}
+
+// Currently VS Code uses a static back button for every QuickInput, so we can't redefine any of its
+// properties without potentially affecting other extensions. Creating a wrapper is possible, but it
```

Comment:
The comment describes the issue of patching the button directly. The button is a static instance so I'm not entirely sure if each individual extensions gets their own copy, or if it's shared across extensions. If it's shared, then modifying it could have undefined behavior.